ics23.go/proof.go

```diff
@@ -0,0 +1,167 @@
+package iavl
```

Comment:
This probably shouldn't be named `iavl` when the root package is also called `iavl`. Let's either move this into the root, or rename it.

Reply:
Why not just place these files in the top level package?

Rename `ics23.go/proof.go` => `proof_ics23.go` or such
And `test_helpers.go` can be merged into `proof_ics23_test.go` (as it doesn't seem to be used otherwise.

2 files. Simple.

Reply:
Yeah, let's move them to the root, then we can merge this and start integrating with the main tree.